sw/device/lib/pinmux.c

```diff
@@ -60,4 +60,24 @@ void pinmux_init(void) {
   reg_offset = kTopEarlgreyPinmuxMioOutIoc11 << 2;
   mask = PINMUX_MIO_OUTSEL_0_OUT_0_MASK;
   mmio_region_write32(reg32, reg_offset, reg_value & mask);
+
+  // Configure UART1 RX input to connect to MIO pad IOC8
+  reg32 = mmio_region_from_addr(PINMUX0_BASE_ADDR +
+                                PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET);
+  reg_value = kTopEarlgreyPinmuxInselIoc8;
+  // We've got one insel configuration field per register. Hence, we have to
+  // convert the enumeration index into a byte address using << 2.
+  reg_offset = kTopEarlgreyPinmuxPeripheralInUart1Rx << 2;
+  mask = PINMUX_MIO_PERIPH_INSEL_0_IN_0_MASK;
+  mmio_region_write32(reg32, reg_offset, reg_value & mask);
+
+  // Configure UART1 TX output to connect to MIO pad IOC9
+  reg32 =
+      mmio_region_from_addr(PINMUX0_BASE_ADDR + PINMUX_MIO_OUTSEL_0_REG_OFFSET);
+  reg_value = kTopEarlgreyPinmuxOutselUart1Tx;
+  // We've got one insel configuration field per register. Hence, we have to
+  // convert the enumeration index into a byte address using << 2.
+  reg_offset = kTopEarlgreyPinmuxMioOutIoc9 << 2;
+  mask = PINMUX_MIO_OUTSEL_0_OUT_0_MASK;
+  mmio_region_write32(reg32, reg_offset, reg_value & mask);
```

Comment:
CC @weicaiyang since he is looking into testing muxed peripherals at the top-level. 
We probably have to start defining different pinmux configurations for the FPGA targets and the chip-level testbench...

Reply:
it might be good to bring this up with the software team on how we want to do platform specific inits. 
We have link-time injection now of a specific values, so maybe it would be good to use something similar to figure out what platform we're on and how to diversify some of this stuff.

Reply:
actually thinking about this a bit more, since pinmux_init is really a function that's called by boot_rom right now, the fpga specific code "probably" doesn't belong here. It makes more sense to split this code out into dedicated test cases.

Since this code likely will be re-used across test cases, it makes sense to put this into the util area (@cfrantz suggestion) such that it can be re-used by other test cases. That might be the best thing to do here instead of loading up the library function used by the fake rom.

Reply:
I'm not really familiar with other use cases and configurations but this seems reasonable to me for the additional UART that CW310 has as long as this configuration does not break other targets. If we want to limit this to CW310 only, then we can use `kDeviceType` as @tjaychen mentioned and move this part to a separate function.

Reply:
So right now this probably still works also for other platforms, because it is only adding another UART. 
But as we are adding more ASIC chip-level tests that test more of the peripherals (like all our I2C's, pattgen SPI's etc), these configurations will almost necessarily have to be different at some point.

This does not have to be a blocker for this PR, though.

What would be the best forum to talk about this? SW meeting or one of our silicon meetings?

CC: @sriyerg

Reply:
i think whichever comes up first that people can make.

Reply:
Thanks for bringing this topic up. Having a clean way for specifying such platform specific init stuff is preferable. This could also help to support multiple top levels. Today (and even without this PR), we need to patch the software tree to enable compilation for the CW305/top_englishbreakfast. This is ugly and error prone.